opensrp-reveal/src/main/java/org/smartregister/reveal/util/InteractorUtils.java

```diff
@@ -110,4 +123,88 @@ public boolean archiveClient(String baseEntityId, boolean isFamily) {
+                    JSONObject eventJson = new JSONObject(gson.toJson(event));
+                    eventJson.put("dateVoided", now);
+                    eventJson.put(EventClientRepository.event_column.syncStatus.name(), BaseRepository.TYPE_Unsynced);
+                    taskEvents.put(eventJson);
+                }
+
+                eventClientRepository.batchInsertEvents(taskEvents, 0);
+            }
+
+            // use original task.for in cases such as Case Confirmation where the task.for changes to point to operational area
+            String taskForEntity = StringUtils.isNotEmpty(taskDetails.getTaskEntity()) ? taskDetails.getTaskEntity() : task.getForEntity();
+
+            Event resetTaskEvent = RevealJsonFormUtils.createTaskEvent(taskForEntity, Utils.getCurrentLocationId(),
+                    null, Constants.TASK_RESET_EVENT, Constants.STRUCTURE);
+            JSONObject eventJson = new JSONObject(gson.toJson(resetTaskEvent));
+            eventJson.put(EventClientRepository.event_column.syncStatus.name(), BaseRepository.TYPE_Unsynced);
+
+            populateEventDetails(eventJson, task.getIdentifier(), taskDetails.getBusinessStatus(), taskDetails.getTaskStatus(), taskDetails.getStructureId());
+            eventClientRepository.addEvent(task.getForEntity(), eventJson);
+
+
+        } catch (Exception e) {
+            Timber.e(e);
+            archived = false;
+        }
+
+        return archived;
+    }
+
+    public List<String> getFormSubmissionIdsFromEventTask(SQLiteDatabase db, String taskIdentifier) {
+        List<String> formSubmissionIds = new ArrayList<>();
+        Cursor cursor = null;
+
+        try {
+            String query = String.format("select %s from %s where %s = ?",
+                    BASE_ENTITY_ID, EVENT_TASK_TABLE, TASK_ID);
+            cursor = db.rawQuery(query, new String[]{taskIdentifier});
+
+            while (cursor.moveToNext()) {
+                formSubmissionIds.add(cursor.getString(cursor.getColumnIndex(BASE_ENTITY_ID)));
+            }
+
+        } finally {
+            if (cursor != null) {
+                cursor.close();
+            }
+        }
+
+        return formSubmissionIds;
+    }
+
+    private void populateEventDetails(JSONObject eventJson, String taskIdentifier,
+                                      String taskBusinessStatus, String taskStatus, String structureId){
+        try {
+            JSONObject formData = new JSONObject();
+            formData.put(Constants.Properties.TASK_IDENTIFIER, taskIdentifier);
+            formData.put(TASK_BUSINESS_STATUS, taskBusinessStatus);
+            formData.put(Constants.Properties.TASK_STATUS, taskStatus);
+            formData.put(Constants.Properties.APP_VERSION_NAME, BuildConfig.VERSION_NAME);
+            formData.put(Constants.Properties.LOCATION_ID, structureId);
+            eventJson.put(DETAILS, formData);
+        } catch (JSONException e) {
+            Timber.e(e);
+        }
+    }
```

Comment:
Use the existing method `org.smartregister.reveal.util.RevealJsonFormUtils#populateFormDetails`